libqpdf/qpdf/QPDFObject_private.hh

```diff
@@ -165,6 +160,19 @@ class QPDFObject
         return dynamic_cast<T*>(value.get());
     }
 
+    // Array methods
+
+    int
+    size(bool type_warn = false, bool allow_null = false) const
+    {
+        return value->size(type_warn, allow_null);
+    }
+    QPDFObjectHandle
+    at(int n, bool warn = false, bool allow_null = false) const
```

Comment:
Same as size -- I feel like this is heading a bit in the wrong direction regarding type safety. I'd rather see the clutter in QPDFObjectHandle of asArray(), isArray(), etc. -- it makes it more obvious inside the abstraction layer that's responsible for type checking.